pkg/controller/vsphere/reconciler.go

```diff
@@ -507,3 +512,43 @@ func (vm *virtualMachine) getNetworkStatusList(client *vim25.Client) ([]NetworkS
 
 	return networkStatusList, nil
 }
+
+// IgnitionConfig returns a slice of option values that set the given data as
+// the guest's ignition config.
+func IgnitionConfig(data []byte) []types.BaseOptionValue {
+	config := EncodeIgnitionConfig(data)
+
+	if config == "" {
+		return nil
+	}
+
+	return []types.BaseOptionValue{
+		&types.OptionValue{
+			Key:   "guestinfo.ignition.config.data",
```

Comment:
nit: could move this to constants

Reply:
where are these magic keys documented, can we add a reference here?

Reply:
I'm not sure there's good documentation to link to yet unfortunately. Maybe the old CoreOS docs, but that seems a little weird.

Reply:
Fixed.